accio-main/src/main/java/io/accio/main/connector/duckdb/DuckDBMetadata.java

```diff
@@ -14,47 +14,69 @@
 import io.accio.base.AccioException;
 import io.accio.base.Column;
 import io.accio.base.ConnectorRecordIterator;
 import io.accio.base.Parameter;
 import io.accio.base.client.AutoCloseableIterator;
+import io.accio.base.client.duckdb.CacheStorageConfig;
+import io.accio.base.client.duckdb.DuckDBConfig;
+import io.accio.base.client.duckdb.DuckDBConnectorConfig;
+import io.accio.base.client.duckdb.DuckDBSettingSQL;
 import io.accio.base.client.duckdb.DuckdbClient;
-import io.accio.base.client.duckdb.DuckdbTypes;
+import io.accio.base.config.ConfigManager;
 import io.accio.base.metadata.TableMetadata;
-import io.accio.base.sql.SqlConverter;
+import io.accio.base.type.DateType;
+import io.accio.base.type.PGType;
 import io.accio.base.type.VarcharType;
 import io.accio.cache.DuckdbRecordIterator;
 import io.accio.connector.StorageClient;
 import io.accio.main.metadata.Metadata;
 import io.accio.main.pgcatalog.builder.DuckDBFunctionBuilder;
 import io.accio.main.pgcatalog.builder.PgFunctionBuilder;
-import io.accio.main.wireprotocol.PgMetastore;
+import io.airlift.log.Logger;
 import io.trino.sql.tree.QualifiedName;
 
 import javax.inject.Inject;
 
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.sql.Date;
+import java.time.LocalDate;
 import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
 
 import static io.accio.base.metadata.StandardErrorCode.GENERIC_INTERNAL_ERROR;
 import static io.accio.main.pgcatalog.PgCatalogUtils.ACCIO_TEMP_NAME;
 import static io.accio.main.pgcatalog.PgCatalogUtils.PG_CATALOG_NAME;
-import static java.lang.String.format;
+import static java.util.Locale.ENGLISH;
 import static java.util.Objects.requireNonNull;
 import static java.util.stream.Collectors.toList;
 
 public class DuckDBMetadata
-        implements Metadata, PgMetastore
+        implements Metadata
 {
-    private final DuckdbClient duckdbClient;
-    private final SqlConverter sqlConverter = new DuckDBSqlConverter();
+    private static final Logger LOG = Logger.get(DuckDBMetadata.class);
+    private final ConfigManager configManager;
+    private DuckdbClient duckdbClient;
     private final PgFunctionBuilder pgFunctionBuilder;
+    private final Map<String, String> pgToDuckDBFunctionNameMappings;
+    private final AtomicReference<DuckDBSettingSQL> duckDBSettingSQL = new AtomicReference<>(new DuckDBSettingSQL());
 
     @Inject
-    public DuckDBMetadata(DuckdbClient duckdbClient)
+    public DuckDBMetadata(
+            ConfigManager configManager)
     {
-        this.duckdbClient = requireNonNull(duckdbClient, "duckdbClient is null");
-        this.pgFunctionBuilder = new DuckDBFunctionBuilder(this);
+        this.configManager = requireNonNull(configManager, "configManager is null");
+        initDuckDBSettingSQLIfNeed();
+        this.duckdbClient = buildDuckDBClient();
```

Comment:
We should handle executing Init-SQL or Session-SQL failed when the server startup. We should skip the failed SQL to start the server. Then, users can update the SQL by API.